api/envoy/config/listener/v3/listener.proto

```diff
@@ -119,6 +138,25 @@ message Listener {
   // :ref:`FAQ entry <faq_how_to_setup_sni>`.
   repeated FilterChain filter_chains = 3;
 
+  // A filter chain matcher resolving the filter chain name based on the connection properties.
+  // The following inputs are supported by the matcher:
```

Comment:
More docs here and some examples? Also discuss how various fields would now be ignored/deprecated for review clarity?

Reply:
Added blurb about about not using old field if this is enabled. Will follow up with worked out example once we agree on the structure.